### ISSUE TITLE
Typescript does not identify when a property can be undefined due to a safe operator being used

### ISSUE THREAD

**Issue by joel-thompson** [microsoft/TypeScript#56608]
### 🔎 Search Terms

"safe navigation operator", "optional chaining"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about our issue.

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgPICMBWyDeAoZZABzijgFsAuZAZzClAHMBuPAXzzzAE8iUNMqKADkA9mGQBeNFmQAfZAFcQAEwgxQEFazwAbCBJijR1AULFgdCUSBqj9AOl2jGACiOiA-A5JlyASmZkAHpg5B4+GgQGIgkAaxBRAHcacIALOAlfCmRrRV0VZHQUOgYQRmRRKCVVdU1CtOgIBwc8a1t7Zuc3D29s8gcaXWAkV39AkLCIiCiYiRo00XzC63IiXThQZEaoCAAaYlIc8jhuIpRlNQ0QLWQ4VWREiThaeiYHIA

### 💻 Code

```ts
interface Obj {
  param: string;
}

type ObjOrNot = Obj | undefined;

let foo: ObjOrNot;

console.log(foo?.param); // typescript knows that param could be string or undefined here.
console.log(foo?.param.slice()); // typescript should complain here, param may be undefined and not a string, but it says it is a string.
```

### 🙁 Actual behavior

typescript allows `.slice()` to be called on `param`

### 🙂 Expected behavior

typescript should give an error when calling `.slice()` on `param` because `param` may be undefined because of the safe navigation operator before it. 

### Additional information about the issue

Seems related to but still a bit different from this issue: https://github.com/microsoft/TypeScript/issues/53872

**Comment by MartinJohns** [comment]
When `foo` is `undefined`, then the entire right side expression won't be called. That means `param.slice()` is never evaluated.

Try running your playground. It works just fine, even when `foo` is `undefined`.

**Comment by fatcerberus** [comment]
> typescript knows that param could be string or undefined here.

Correction: It knows that the *whole expression* `foo?.param` can be `string` or  `undefined`.  However, `?.` is short-circuiting so if `foo` is defined, `foo.param` is always a string, per the types.  If `foo` is nullish then `foo?.param.slice` doesn't even get evaluated.

**Comment by RyanCavanaugh** [comment]
The behavior of `?.` is admittedly counterintuitive; the expression `a?.b.c` is not the same as `(a?.b).c`

Regardless, TypeScript accurately models the runtime behavior here.

**Comment by joel-thompson** [comment]
interesting, apologies for mix up. I could have sworn I've run into issues around `Cannot read properties of undefined` similar to my example, but I can't reproduce it now so it must have been something else!
